Trust main

Omnione opened the pull request:
Trust: This creates vector list of trust id's and query's the id's then generates a new g_PTrustList. just like the g_PPetlist, that can iterated through using the trustID so you can call the trust using the spellid rather than the old way of adding a massive amount of magic numbers in pet_list.sql or somthing like that.

The only real logic that i changed was the order of adding trust to the list of spawned trust so they follow each other from first spawned onwards down the list.

fixed some heap errors due to the tick still running after despawn.

No combat changes or stat changes have been made, you still can't cast on trust and trust wont cast or perform weapon skills/abilities either.

Omnione commented:
man i keep forgetting to change the title, lol

Omnione commented:
wtf, this builds fine for me, hmm

Omnione commented:
Ok 3rd time lucky

Omnione commented:
If this last commit dont work can somone else take a look and find out what the hell im doing wrong please

TeoTwawki commented on src/map/entities/charentity.h:
```diff
@@ -213,19 +213,20 @@ class CCharEntity : public CBattleEntity
+    uint16			  m_asaCurrent;					// current mission of A Shantotto Ascension
 
-    // TODO: половина этого массива должна храниться в char_vars, а не здесь, т.к. эта информация не отображается в интерфейсе клиента и сервер не проводит с ними никаких операций
+    // TODO: half of this array should be stored in char_vars, and not here, because this information is not displayed in the client interface and the server does not perform any operations with them
```
Kill that comment entirely - f charvars

Omnione replied:
all done.

Omnione commented on scripts/globals/spells/trust/naji.lua:
```diff
@@ -0,0 +1,16 @@
+end
+
+function onSpellCast(caster,target,spell)
+    -- Todo: put Trusts's spawn in dialog here or in entity script?
```
all done.

Omnione commented:
I have added the addition of a global trust.lua just incase we need it later on for something and made all the trust scripts require it. and fixed the fact i left out the changes to LoadTrustlist in map.ccp
